File: LMIS/LMIS.Web/Modules/BookManage/Categories/CategoriesColumns.cs
using Serenity.ComponentModel;
using System;
using System.ComponentModel;

namespace LMIS.BookManage.Columns;

[ColumnsScript("BookManage.Categories")]
[BasedOnRow(typeof(CategoriesRow), CheckNames = true)]
public class CategoriesColumns
{
    [EditLink,  AlignRight]
    public long CategoryId { get; set; }
    [EditLink]
    public string CategoryName { get; set; }
    [Width(150)]
    [DateTimeFormatter(DisplayFormat = "yyyy/MM/dd HH:mm:ss")]
    public DateTime CreateTime { get; set; }
    [Width(150)]
    [DateTimeFormatter(DisplayFormat = "yyyy/MM/dd HH:mm:ss")]
    public DateTime UpdateTime { get; set; }
}

File: LMIS/LMIS.Web/Modules/ServerTypes/BookManage/CategoriesColumns.ts
import { ColumnsBase, fieldsProxy } from "@serenity-is/corelib";
import { Column } from "@serenity-is/sleekgrid";
import { CategoriesRow } from "./CategoriesRow";

export interface CategoriesColumns {
    CategoryId: Column<CategoriesRow>;
    CategoryName: Column<CategoriesRow>;
    CreateTime: Column<CategoriesRow>;
    UpdateTime: Column<CategoriesRow>;
}

export class CategoriesColumns extends ColumnsBase<CategoriesRow> {
    static readonly columnsKey = 'BookManage.Categories';
    static readonly Fields = fieldsProxy<CategoriesColumns>();
}